### PR TITLE
Bring up OpenGApps to support Android 9.0 (1/4)

### PR DESCRIPTION
**Pull request by SpasilliumNexus**
Changes:

**Enable OpenGApps support for Android 9.0 Pie**

**Create symlink for SetupWizard on Android 9.0 (ARM64)**

**Limit SetupWizard library to Android 9.0 (ARM64)**
- This library only has an ARM64 variant. No info is known on exactly what it does, but we'll include it just in case to avoid any issues.

**Add new and updated permissions, sysconfig from Android 9.0**

**Only com.google.android.dialer.support.xml is in Android 9.0**
- dialer_experience.xml is not needed anymore since this file has the same permissions and is redundant with 9.0

**Add additional sysconfigs for Pixel devices**
- These files contain additional permissions and features exclusively for Pixel

**Add additional sysconfig for enabling Google exclusives on non-Google devices**
- Some Pixel features such as Digital Wellbeing and exclusive Wallpaper categories work with no problem on non-Google devices

**Remove uneeded 2017 camera permission and jar**
- To this day these files still have issues with launching the Google Camera app

**Install legacy GMSSetup only on Android 8.0 & 8.1**
- This is no longer included in Android 9.0

**Play Services folder name is PrebuiltGmsCorePi on Android 9.0**

**Add & separate Android Platform Services between Android 8.0 & 9.0**
- There are two separate package names between the versions. 8.0 using "com.google.android.gms.policy_sidecar_o" and 9.0 using "com.google.android.gms.policy_sidecar_aps". With Android 9.0, not having this app brought random FCs about it being missing during SetupWizard.

**Add Data Transfer Tool to Android 9.0 (Core)**
- Not having this included prevents users from restoring their device's apps, settings, etc in Android 9.0. **_Moved to Core in a later commit._**

**Add Actions Services to Android 9.0 (Super/Aroma)**
- This app is what enables text selection while in the overview/recents screen.

**Add Google Sound Picker to Android 9.0 (Super/Aroma)**
- Not having this app prevented users from selecting alarms, ringtones, and/or notification sounds in apps such as Google Clock. Also, if this installed, Android 9.0 uses it by default. **_Moved to Core in a later commit._**

**Add Better Together to Android 9.0 (Super/Aroma)**
- This app allows integration with Chrome OS such as Android Messages and Instant Tethering.

**Add Digital Wellbeing to Android 9.0 (Super/Aroma)**
- This app allows the user to set and monitor their device usage. It also works on devices other than a Pixel.

**Add Markup to Android 9.0 (ARM64) (Super/Aroma)**
- This app allows the user to highlight and draw on their screenshots by tapping edit below the screenshot in the notification pulldown. Working only on ARM64 devices due to the required library that's used with the app. **_Moved to Core in a later commit._**

**Fix indentation automatically created with Notepad++**

**Remove duplicate statement in installer, fix typo in Aroma**

**Search in more locations for gapps-config (/data)**
- In addition to the current locations, /data was added for those who create "auto-installers" that copy a gapps-config file to the root of that location.

**Apply correct permission to build.prop on Android 8.0+**
- Not having this permission applied causes some devices to not boot.

**Repair missing check entries in inc.aromadata.sh**

**Move Google Sound Picker to Core**
- See "Add Google Sound Picker to Android 9.0 (Super/Aroma)"

**GoogleLoginService is no longer available in Android 8.0+**
- GoogleServicesFramework takes place of this app since 8.0

**Move Digital Wellbeing to Core**
- See "Add Digital Wellbeing to Android 9.0 (Super/Aroma)"

**Move Markup to Core**
- See "Add Markup to Android 9.0 (ARM64) (Super/Aroma)"

**Installer: Replace all instances of "insure" to "ensure"**

**Compatibility: Replace all instances of "insure" to "ensure"**

### CODE REVIEW

**Comment by SpasilliumNexus** [comment]
**_NOTES:_**

The only current issue with this bringup is some ARM devices cannot get past signing into their Google Account, whether it be during SetupWizard, or a Google app that requires signing in on Android 9.0.

I've given/suggested every kind of potential solution with solving the problem that I can think of, from renaming/deleting apps, to going back to Oreo core files, but nothing solved it.

The two known devices to suffer from this are the Lenovo A6000 (phone), and Xiaomi MiPad 1 (tablet).

It should be known that a report of the 2013 Moto G, also an ARM device, works just fine.

I have attached logcats of both devices below in case anyone wants to see what's going on, and (hopefully) have a solution.

**Comment by SpasilliumNexus** [comment]
[lenovo_a6000_logcat.txt](https://github.com/opengapps/opengapps/files/2342455/lenovo_a6000_logcat.txt)

[xiaomi_mipad1_logcat.txt](https://github.com/opengapps/opengapps/files/2342456/xiaomi_mipad1_logcat.txt)

**Comment by nezorflame** [comment]
+1 to that.
@SpasilliumNexus did a tremendous amount of work.
@mfonville @NicholasBuse any news on merging this and https://github.com/opengapps/all/pull/24?

**Comment by NicholasBuse** [comment]
Yes Tommy did! Thanks!

I spent some time yesterday building legacy builds pre- and post- Tommy's PRs. The 9.0 stuff looks good, I'm just checking older builds, especially when files (*.xml) go away...

I hope to complete today and ping @mfonvile to update the build server.

**Comment by nezorflame** [comment]
@NicholasBuse thanks, much appreciated!
If you guys would need more time with the build server, just ping me - I can continue making unofficial ones on my build server until yours are ready.

File: scripts/inc.compatibility.sh [hunk 1]
```diff
@@ -509,9 +509,11 @@ photosvrmode"
 }
 
 api25hack(){
-  if [ "$API" -ge "25" ]; then
+  if [ "$API" -eq "25" ]; then
     gappscore="$gappscore
-gmssetup"
```
Comment (NicholasBuse) [inline scripts/inc.compatibility.sh]: @SpasilliumNexus, I assume that that the removal of 'gmssetup' here should be added in api26hack() and api27hack() below?

Comment (NicholasBuse) [reply scripts/inc.compatibility.sh]: I am going to merge this and the "all" one.  I'll fix this when I hear from you (or not if it is correct).  I have one other change to opengapps/opengapps to make as well

Comment (NicholasBuse) [reply scripts/inc.compatibility.sh]: Commited: https://github.com/opengapps/opengapps/commit/c843fe13f77b5a19cdd2a91bec65ff54280f171e

**Comment by NicholasBuse** [comment]
9.0 builds coming tonight...

Thanks @SpasilliumNexus!

File: scripts/inc.compatibility.sh (hunk 1 above)
Comment (SpasilliumNexus) [reply scripts/inc.compatibility.sh]: Hey sorry about not getting back. I didn't have Gmail set up for notifications on my "dev" account after a clean ROM install >_<

But yes, that's where I mean to add it at. Thank you for the fix!

**Comment by NicholasBuse** [comment]
I lied ...
... But they are coming tonight! (Build Server only got partially updated and did not execute 9.0 builds)

**Pull request merged**